src/TopoMojo.Hypervisor/Proxmox/ProxmoxVlanManager.cs

```diff
@@ -0,0 +1,228 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Caching.Memory;
+using Microsoft.Extensions.Logging;
+using TopoMojo.Hypervisor.Common;
+using TopoMojo.Hypervisor.Proxmox.Models;
+
+namespace TopoMojo.Hypervisor.Proxmox
+{
+    public interface IProxmoxVlanManager
+    {
+        Task<IEnumerable<PveVnet>> DeleteVnets(IEnumerable<string> vnetNames);
+        Task<IEnumerable<PveVnet>> DeleteVnetsByTerm(string term);
+        Task<IEnumerable<PveVnet>> GetVnets();
+        Task<bool> HasNetwork(string networkName);
+        Task<bool> HasNetworks(IEnumerable<string> networkNames);
+        Task<IEnumerable<PveVnet>> Provision(IEnumerable<string> vnetNames);
+    }
+
+    public class ProxmoxVlanManager : IProxmoxVlanManager
+    {
+        private readonly static Lazy<SemaphoreSlim> _deploySemaphore = new Lazy<SemaphoreSlim>(() => new SemaphoreSlim(1));
+        // defaults to a debounce period of 300ms, but can be changed using the `Pod__Vnet__ResetDebounceDuration`. A maximum
+        // debounce can be set using `Pod__VNet__ResetDebounceMaxDuration`.
+        private readonly static Lazy<DebouncePool<PveVnetOperation>> _vnetOpsPool = new Lazy<DebouncePool<PveVnetOperation>>(() => new DebouncePool<PveVnetOperation>(300));
+        private readonly static IMemoryCache _recentVnetOpsCache = new MemoryCache(new MemoryCacheOptions { });
+
+        private readonly int _cacheDurationMs = 600;
```

Comment:
Since this value is always changed in the constructor, maybe set this to 0 or make it nullable and set to null so no one looks at it and thinks it's actually 600ms?